clients/src/main/java/org/apache/kafka/common/requests/MetadataResponse.java

```diff
@@ -334,23 +338,27 @@ public boolean isInternal() {
 
     }
 
+    // TODO: rename this class to MetadataResponsePartitionState
```

Comment:
@ijuma It may be useful to rename this class such that, instead of doing `new MetadataResponse.PartitionMetadata(...)` in the code, we can do `import org.apache.kafka.common.requests.MetadataResponse.PartitionMetadata` and `new PartitionMetadata(...)` e.g. in `MetadataCache.java`. The latter is more concise.

Anyway, I have replaced this TODO with just a comment.